Makefile

```
@@ -45,39 +45,39 @@ dockerise: build-voter build-ballot build-ecserver build-ec build-test
 .PHONY: build-ballot
 build-ballot:
 ifdef DOCKER_HOST
-	docker build -H ${DOCKER_HOST} -t ${BALLOT_IMG}:${IMAGE_TAG} -f ballot/Dockerfile ballot
+	docker -H ${DOCKER_HOST} build -t ${BALLOT_IMG}:${IMAGE_TAG} -f ballot/Dockerfile ballot
 else
 	docker build -t ${BALLOT_IMG}:${IMAGE_TAG} -f ballot/Dockerfile ballot
 endif
```

It's better to setup ${DOCKER_HOST_ARG} once around top of the script and use in necessary targets.

```
DOCKER_HOST_ARG=""
ifdef DOCKER_HOST
        DOCKER_HOST_ARG="-H ${DOCKER_HOST}"
endif
```

```suggestion
	docker ${DOCKER_HOST_ARG} build -t ${BALLOT_IMG}:${IMAGE_TAG} -f ballot/Dockerfile ballot
```